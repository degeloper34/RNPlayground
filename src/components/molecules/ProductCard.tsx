import {
  Dimensions,
  Image,
  Pressable,
  View,
  StyleSheet,
  StyleProp,
  ViewStyle,
} from "react-native";
import Colors from "../../constants/Colors";
import {CustomText, FlexView} from "../atoms";

interface IProps {
  title: string;
  imageUrl: string;
  price: number;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}

export function ProductCard({title, imageUrl, price, onPress, style}: IProps) {
  const {imageStyle, innerViewStyle, btnStyle} = styles;

  return (
    <Pressable onPress={onPress} style={[btnStyle, style]}>
      <FlexView flex={3}>
        <Image
          source={{uri: imageUrl}}
          style={imageStyle}
          resizeMode={"contain"}
        />
      </FlexView>

      <View style={innerViewStyle}>
        <FlexView flex={2}>
          <CustomText
            text={title}
            type={"medium"}
            textColor={Colors.mainGrey}
            fontSize={12}
            numberOfLines={2}
          />
        </FlexView>

        <FlexView>
          <CustomText
            text={"$" + price}
            type={"bold"}
            textColor={Colors.mainYellow}
            fontSize={12}
            numberOfLines={2}
          />
        </FlexView>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  btnStyle: {
    width: Dimensions.get("screen").width / 3,
    height: Dimensions.get("screen").height / 3.5,
    borderRadius: 8,
    backgroundColor: "white",
    overflow: "hidden",
  },
  imageStyle: {
    flex: 1,
  },
  innerViewStyle: {
    flex: 1,
    padding: 3,
    backgroundColor: Colors.mainNight,
    marginTop: 12,
  },
});
