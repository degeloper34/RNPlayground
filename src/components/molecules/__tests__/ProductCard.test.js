import renderer from "react-test-renderer";
import React from "react";
import {ProductCard} from "../ProductCard";

describe("ProductCard", () => {
  it("renders correctly with mandatory props", () => {
    const onPress = jest.fn();

    const component = renderer
      .create(
        <ProductCard
          title={"Jacket"}
          imageUrl={"https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"}
          price={"120"}
          onPress={onPress}
        />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with mandatory props and ProductListScreen style", () => {
    const onPress = jest.fn();

    const component = renderer
      .create(
        <ProductCard
          title={"Jacket"}
          imageUrl={"https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"}
          price={"120"}
          onPress={onPress}
          style={{marginRight: 12}}
        />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with mandatory props and CategoryScreen style", () => {
    const onPress = jest.fn();

    const component = renderer
      .create(
        <ProductCard
          title={"Jacket"}
          imageUrl={"https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"}
          price={"120"}
          onPress={onPress}
          style={{flex: 1, margin: 12}}
        />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
