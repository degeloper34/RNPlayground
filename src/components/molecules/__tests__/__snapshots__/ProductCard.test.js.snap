// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly with mandatory props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "height": 381.14285714285717,
        "overflow": "hidden",
        "width": 250,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#21325E",
        "flex": 1,
        "marginTop": 12,
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F0F0F0",
              "fontFamily": "sora-medium",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Jacket
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F1D00A",
              "fontFamily": "sora-bold",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        $120
      </Text>
    </View>
  </View>
</View>
`;

exports[`ProductCard renders correctly with mandatory props and CategoryScreen style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "height": 381.14285714285717,
        "overflow": "hidden",
        "width": 250,
      },
      Object {
        "flex": 1,
        "margin": 12,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#21325E",
        "flex": 1,
        "marginTop": 12,
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F0F0F0",
              "fontFamily": "sora-medium",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Jacket
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F1D00A",
              "fontFamily": "sora-bold",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        $120
      </Text>
    </View>
  </View>
</View>
`;

exports[`ProductCard renders correctly with mandatory props and ProductListScreen style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "height": 381.14285714285717,
        "overflow": "hidden",
        "width": 250,
      },
      Object {
        "marginRight": 12,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#21325E",
        "flex": 1,
        "marginTop": 12,
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F0F0F0",
              "fontFamily": "sora-medium",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Jacket
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          Array [
            undefined,
            Object {
              "color": "#F1D00A",
              "fontFamily": "sora-bold",
              "fontSize": 12,
              "textDecorationLine": "none",
            },
          ]
        }
      >
        $120
      </Text>
    </View>
  </View>
</View>
`;
